dm/syncer/syncer.go

```diff
@@ -172,7 +172,10 @@ type Syncer struct {
 	exprFilterGroup *ExprFilterGroup
 	sessCtx         sessionctx.Context
 
-	closed atomic.Bool
+	// int value of pb.Stage, stage diagram:
+	// new --> running --> stopped --> finished
+	//     |-> finished
+	stage atomic.Int32
```

Comment:
what is stage of `finished` ? syncer will never finished, so seems use one var `isRunning` is enough?

Reply:
`isRunning` overlaps with existing `isClosed` fields, i.e. running means not closed, so i merge them into one fields called stage(reusing existing enum), `finished` is set when `Close` is called.

Reply:
will revert and add `isRunning` as other routine(ungraceful kill) may change stage and it's not waiting `s.Process`(no one manages this routine) exist, so stage may leave with a `stopped` value not `finished`. i can move set stage to running/stopped part to a routine managed by `syncer.runWg`, but it looks odd